# junction
# 0 - T Junction
# 1 - X - junction
int32 junction

# 2 : left
# 1 : front
# 0 : right

int32 turn_direction

# distance to travel after turning at junction
float32 distance
---
---
int32 state